test/unit/api/cpp/solver_black.cpp

```diff
@@ -2524,10 +2524,60 @@ TEST_F(TestApiBlackSolver, pluginUnsat)
 {
   PluginUnsat pu(d_tm);
   d_solver->addPlugin(pu);
+  ASSERT_TRUE(pu.getName() == "PluginUnsat");
   // should be unsat since the plugin above asserts "false" as a lemma
   ASSERT_TRUE(d_solver->checkSat().isUnsat());
 }
 
+class PluginListen : public Plugin
+{
+ public:
+  PluginListen(TermManager& tm)
+      : Plugin(tm),
+        d_tm(tm),
+        d_hasSeenTheoryLemma(false),
+        d_hasSeenSatClause(false)
+  {
+  }
+  virtual ~PluginListen() {}
+  void notifySatClause(const Term& cl) override { d_hasSeenSatClause = true; }
+  bool hasSeenSatClause() const { return d_hasSeenSatClause; }
+  void notifyTheoryLemma(const Term& lem) override
+  {
+    d_hasSeenTheoryLemma = true;
+  }
+  bool hasSeenTheoryLemma() const { return d_hasSeenTheoryLemma; }
+  std::string getName() override { return "PluginListen"; }
+
+ private:
+  /** Reference to the term manager */
+  TermManager& d_tm;
+  /** have we seen a theory lemma? */
+  bool d_hasSeenTheoryLemma;
+  /** have we seen a SAT clause? */
+  bool d_hasSeenSatClause;
+};
+
+TEST_F(TestApiBlackSolver, pluginListen)
+{
+  PluginListen pl(d_tm);
+  d_solver->addPlugin(pl);
```

Comment:
Doesn't this have to enable option `plugin-notify-sat-clause-in-solve`?

Reply:
No, `plugin-notify-sat-clause-in-solve` means "only inform me of SAT clauses during solving", whereas if its off, then you get notified of SAT clauses during initialization as well.

For some reason, for this unit test, the *only* clauses it is notified are prior to solving (despite the *.smt2 equivalent notifying SAT learned clauses during solving).  I need to investigate this further.